File: content/atlas/source/includes/fts/materialized-view/CreateIndexExample.java
import com.mongodb.client.*;
import org.bson.Document;

public class CreateIndexExample {
    public static void main(String[] args) {
        // Replace the connection string with your MongoDB deployment's connection string
        String connectionString = "<connection-string>";
        
        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("sample_supplies");
            MongoCollection<Document> collection = database.getCollection("monthlyPhoneTransactions");
            String indexName = "monthlySalesIndex";

            // Create the MongoDB Search index definition for the document field
            Document searchIdx = new Document(
                    "mappings",
                    new Document("dynamic", true)
            );
            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}

File: content/atlas/source/includes/fts/materialized-view/create-index-example.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_supplies"]
collection = database["monthlyPhoneTransactions"]

# Create the MongoDB Search index definition for the document field
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "dynamic": True
        },
    },
    name="monthlySalesIndex",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")
